### ISSUE TITLE
Problem with log config files.

### ISSUE THREAD

**Issue by sbernard31** [eclipse/leshan#473]
Currently we face several issues with log config files.

We currently use `logback-test.xml` for tests in almost all modules and `logback.xml` for executable modules.
The way logback load is described [here](https://logback.qos.ch/manual/configuration.html#auto_configuration). In our case it search in classpath for `logback-test.xml` then `logback.xml`.

**Problems in development :** (using eclipse)
1.1) eclipse does not do the difference between `test` classpath and  `runtime` classpath like maven. That's means that if you have both `logback-test.xml` and `logback.xml`, the test one will always be used even if you does not run tests in eclipse.
1.2) if there is several `logback.xml`  or several  `logback-test.xml` files in classpath there is no way to control which one will be used.

**Problems at build :** (using maven)
2) Separation between `test` and `runtime` works well with maven, but if you create an uber jar (jar with dependencies) and there is several logback.xml files, actually we do not control the one we are using.

**My ideas :**
To solve the 1.1) and 1.2) the only solution I see is to move log config file out of all classpath and define specifically which config file to use.
Logback allow to use a system properties to define which config file to use instead of searching in class path.
```
java -Dlogback.configurationFile=/path/to/config.xml chapters.configuration.MyApp1
```
Eclipse users could set it but I would like a solution which works out of the box.
A solution could be to
- move `./src/main/resource/logback.xml` to "./logback-config.xml" (so not in classpath)
- define a static block for all executable module which [define the default value for this properties](https://logback.qos.ch/manual/configuration.html#configFileProperty) using :
```java
static {
    // must be set before the first call to  LoggerFactory.getLogger();
    System.setProperty("logback.configurationFile", "logback-config.xml");
}  
```
If we want to also use this for Junit, we probably need to do move `logback-test.xml` out of the classpath too (as module root folder for example)  and to add this static block for all test classes...
About maven, `logback-test.xml` is no more in the classpath, so we need to add :
```xml
<artifactId>maven-surefire-plugin</artifactId>
<configuration>
<systemPropertyVariables>
	<logback.configurationFile>logback-test.xml</logback.configurationFile>
</systemPropertyVariables>
...
```
For uber-jar, with this method we have no more default config in jar but put a logback-config.xml file in working directory is enough to use it. (no configuration needed)
So we can :
a) add the "logback-config.xml" in output build directory (`target` directory) near the uber-jar, using :
```xml
<build>
<resources>
	<resource>
	<directory>${project.basedir}</directory>
	<targetPath>${project.build.directory}</targetPath>
	<includes>
		<include>logback-config.xml</include>
	</includes>
	</resource>
</resources>
```
b) we can also bring a default `logback.xml` in the uber-jar. It will be used only if `logback.configurationFile` does not exist. Using maven assembly it will looks like this: 
```xml
<assembly xmlns="http://maven.apache.org/ASSEMBLY/2.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.0.0 http://maven.apache.org/xsd/assembly-2.0.0.xsd">
	
	<id>jar-with-dependencies</id>
	<formats>
		<format>jar</format>
	</formats>
	<includeBaseDirectory>false</includeBaseDirectory>
	<files>
		<file>
			<source>logback-config.xml</source>
			<destName>logback.xml</destName>
		</file>
	</files>
	<dependencySets>
		<dependencySet>
			<outputDirectory>/</outputDirectory>
			<useProjectArtifact>true</useProjectArtifact>
			<unpack>true</unpack>
			<scope>runtime</scope>
		</dependencySet>
	</dependencySets>
</assembly>
```
Why we do not face 2)  ? [here](https://maven.apache.org/plugins/maven-assembly-plugin/descriptor-refs.html#jar-with-dependencies) is the `jar-with-dependencies` descriptors.
It uses the attribute `useProjectArtifact` which includes the current project as `dependencySet`, so all duplicate files have the same priority and so no guarantee about which files will be used. But as we use a `file` for logback.xml we will ensure to always have the current module version because `fileItem` have priority on `dependencySet` (for more [[details](https://maven.apache.org/plugins/maven-assembly-plugin/advanced-descriptor-topics.html#Archive_file_resolution)) 

Another good resource about maven build : https://books.sonatype.com/mvnref-book/reference/index.html

I will create a PR with all this modification.

**Comment by sbernard31** [comment]
Integrated in master (commit 8ac82e8ec6815ca0a04cc936efaf15a5d1beed65)
I prepare another PR about cleaning more maven stuff

**Comment by sbernard31** [comment]
It should be fixed by #475.
